File: app/src/main/java/at/shockbytes/coins/storage/StorageManager.java
package at.shockbytes.coins.storage;

import java.util.List;

import at.shockbytes.coins.currency.OwnedCurrency;
import at.shockbytes.coins.network.model.PriceConversion;
import rx.Observable;

/**
 * @author Martin Macheiner
 *         Date: 16.06.2017.
 */

public interface StorageManager {

    void storeOwnedCurrency(OwnedCurrency currency);

    void removeOwnedCurrency(OwnedCurrency currency);

    Observable<List<OwnedCurrency>> loadOwnedCurrencies();

    void updateConversionRates(OwnedCurrency c, List<PriceConversion> conversions);

}

File: app/src/main/java/at/shockbytes/coins/storage/StorageManager.kt
package at.shockbytes.coins.storage

import at.shockbytes.coins.currency.Currency
import at.shockbytes.coins.currency.conversion.PriceConversion
import io.reactivex.Observable

/**
 * @author Martin Macheiner
 * Date: 16.06.2017.
 */

interface StorageManager {

    fun storeCurrency(currency: Currency)

    fun removeCurrency(currency: Currency)

    fun cashoutCurrency(currency: Currency)

    fun loadCurrencies(isCashedOut: Boolean): Observable<List<Currency>>

    fun updateConversionRate(c: Currency, conversions: List<PriceConversion>)

    fun updateBoughtDateCurrency(currency: Currency, boughtDate: Long)

    fun getOwnedCurrencyById(id: Long): Currency

    fun partialCashout(currency: Currency, amountToPayout: Double)
}
